WordPress/src/main/java/org/wordpress/android/ui/posts/HistoryListFragment.kt

```diff
@@ -115,7 +119,32 @@ class HistoryListFragment : Fragment() {
 
         viewModel.listStatus.observe(this, Observer { listStatus ->
             if (isAdded && view != null) {
-                swipeToRefreshHelper.isRefreshing = listStatus == FETCHING
+                swipeToRefreshHelper.isRefreshing = listStatus == HistoryListStatus.FETCHING
+            }
+            when (listStatus) {
+                HistoryListStatus.DONE -> {
+                    updatePostOrPageEmptyView()
+                }
+                HistoryListStatus.FETCHING -> {
+                    actionable_empty_view.title.setText(R.string.history_fetching_revisions)
+                    actionable_empty_view.subtitle.visibility = View.GONE
+                    actionable_empty_view.button.visibility = View.GONE
+                    reloadList(viewModel.revisions.value ?: emptyList())
```

Comment:
How would you feel about removing `reloadList(..)` from `viewModel.listStatus.observe` and adding `_revisions.value = emptyList()` into `HistoryViewModel.create(..)`? It feels a bit weird we need to manually update the list from the view layer even though we are already observing for its changes. Wdyt?

Reply:
The problem with resetting the `_revisions.value` is when you have some data cached and then do a PTR while offline, this will shows the empty screen instead of keeping the cached data.

Reply:
I'm not suggesting to reset in on `onRevisionsFetched`, but set it to `emptyList` in `vm.create(..)`. 

We'd basically set it to emptyData when the VM is created and we'd update it only after we successfully fetch a new data.

I'm sorry if I'm missing something, but I'd like to clarify it before I merge it;).

Reply:
🤦‍♂ I misread your first comment - yes that makes sense, I'll change that.